firebase_admin/project_config_mgt.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Firebase project configuration management module.
+
+This module contains functions for managing various project-level configurations within a
+Firebase project, such as multi-factor authentication (MFA) settings.
+"""
+
+import requests
+
+import firebase_admin
+from firebase_admin import _auth_utils
+from firebase_admin import _http_client
+from firebase_admin import _utils
+from firebase_admin.multi_factor_config_mgt import MultiFactorConfig
+
+_PROJECT_MGT_ATTRIBUTE = '_project_mgt'
+
+__all__ = [
+    'Project',
+
+    'get_project',
+    'update_project',
+]
+
+def get_project(app=None):
+    """Gets the project corresponding to the given project_id.
+
+    Args:
+        app: An App instance (optional).
+
+    Returns:
+        Project: A project object.
+
+    Raises:
+        ValueError: If the project ID is None, empty or not a string.
+        ProjectNotFoundError: If no project exists by the given ID.
+        FirebaseError: If an error occurs while retrieving the project.
+    """
+    project_mgt_service = _get_project_mgt_service(app)
+    return project_mgt_service.get_project()
+
+
+def update_project(mfa=None, app=None):
+    """Update the Project with the given options.
+
+    Args:
+        mfa: Updated Multi Factor Authentication configuration
+                    (optional)
+        app: An App instance (optional).
+
+    Returns:
+        Project: The updated project object.
+
+    Raises:
+        ValueError: If any of the given arguments are invalid.
+        ProjectNotFoundError: If no project exists by the given ID.
+        FirebaseError: If an error occurs while updating the project.
+    """
+    project_mgt_service = _get_project_mgt_service(app)
+    return project_mgt_service.update_project(mfa=mfa)
+
+
+def _get_project_mgt_service(app):
+    return _utils.get_app_service(app, _PROJECT_MGT_ATTRIBUTE, _ProjectManagementService)
+
+
+class Project:
+    """Represents a project in an application.
+    """
+
+    def __init__(self, data):
+        if not isinstance(data, dict):
+            raise ValueError(
+                'Invalid data argument in Project constructor: {0}'.format(data))
+        if not 'name' in data:
+            raise ValueError('Project response missing required keys.')
+
+        self._data = data
+
+    @property
+    def project_id(self):
+        return self._data.get('name')
```

Comment:
'name' response of GET  /v2/projects/project-id/config contains "projects/project-id/config", so if we need the project_id, we should parse it, right?

Reply:
Or you can call this projectConfigPath and change PROJECT_CONFIG_MGT_URL = 'https://identitytoolkit.googleapis.com/v2/projects' to `base_url = 'https://identitytoolkit.googleapis.com/v2'`

Reply:
I don't think we should have 'name' in the project response config anyway. The project_id is passed in as a field in the credentials. Thanks for catching this. Fixed it.